docs/howto/manage-charms.md

```diff
@@ -20,15 +27,51 @@ configures development tooling.
 TODO: Add a reference link in charmcraft for the link above and the 'runtime details' one below, and switch over to external refs.
 -->
 
+## Develop your charm
+
+The Charmcraft profile has configured some commands to help you develop your charm:
+
+- Run `tox` to format and lint the code, and run static type checking and the
+  charm unit tests.
+- Run `tox -e integration` to run the charm integration tests.
+- Run `tox list` to see the available commands.
+
+```{admonition} Best practice
+:class: hint
+
+All charms should provide the commands configured by the Charmcraft profile, to
+allow easy testing across the charm ecosystem. It's fine to tweak the
+configuration of individual tools, or to add additional commands, but keep the
+command names and meanings that the profile provides.
+```
+
+The following tools can also be useful during development:
+
+- To prepare an environment for running integration tests, such as in continuous integration, use [`concierge`](https://github.com/jnsgruk/concierge) or [`actions-operator`](https://github.com/charmed-kubernetes/actions-operator).
+- The [`charming-actions`](https://github.com/canonical/charming-actions) repository includes actions to ensure that libraries are up-to-date, publish charms and libraries, and more.
+
+```{admonition} Best practice
+:class: hint
+
+The quality assurance pipeline of a charm should be automated using a
+continuous integration (CI) system.
```

Comment:
That's kind of obvious, isn't it?

Reply:
All of these are migrated from the big list in Juju, so it must have not been obvious enough at some point. I don't think anyone at Canonical would need to be told this. I'm less sure about external charmers. Some orgs are surprisingly outdated in their practices (although it seems like those orgs would not be the ones writing charms and using Juju). The only feedback I've seen on this is that external orgs already find it hard to know what is required for a mature (listable) charm, so explicit does seem good.

Maybe this would be better saying something along the lines of "your CI should run at least x,y,z", so it implies there is a CI pipeline but is more about what it should be doing.